frontend/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM node:18-bullseye
+
+# Just install the packages; the source code will be loaded in later.
+RUN mkdir /opt/frontend
+WORKDIR /opt/frontend
+
+ADD ./package.json /opt/frontend/
+RUN yarn install --production=false
+
+# individual files in the parent directory need to be added here
+ADD ./vite.config.js /opt/frontend/
+ADD ./index.html /opt/frontend/
```

Comment:
We also have a `.env` file with API tokens that is not tracked by git but seems like we need to add it here.

Reply:
It's located in the same place as `index.html`, etc.

Reply:
Sure!  I've added it in aa2fcdb.

One thing we may want to consider for the future is that we have some other way of sharing secrets.  I think `docker-compose.yml` can support this, or we could put the `.env` file in a directory and mount that into the container.  My concern is that if we add the secret to the container as we are now, that we might accidentally end up publishing the secret along with the container, so making it harder for us to accidentally do that might be wise.

Reply:
@davemfish it appears that `docker-compose.yml` does support mounting secret files into the container at runtime!  But the catch is that it's mounted at `/run/secrets/frontend-api-keys/.env`.  If it's easier, I can see about adding an environment variable about where the `.env` file is located and we can have that set at runtime.

Added this config in 836f4e4.

Reply:
@phargogh thanks for trying that out. We actually rely on `vite`, which relies on `dotenv` to load the `.env`. And it looks like we can set an `envDir` in `vite.config.js` to whatever we want to. So I guess that would work. It just means we can't develop the app outside the context of docker. But I suppose that's the whole point!

Reply:
https://vitejs.dev/config/#envdir